src/config.js

```diff
@@ -0,0 +1,62 @@
+/* @flow */
+import path from 'path';
+
+import requireUncached from 'require-uncached';
+
+import {createLogger} from './util/logger';
+import {UsageError} from './errors';
+
+const log = createLogger(__filename);
+
+type ApplyConfigToArgvParams = {|
+  argv: Object,
+  configObject: Object,
+  defaultValues: Object,
+|};
+
+type ParseConfigParams = {|
+  sourceDir: string,
+  configFileName: string,
+|};
+
+export function applyConfigToArgv({
+  argv,
+  configObject,
+  defaultValues = {},
+}: ApplyConfigToArgvParams): Object {
```

Comment:
This function should not mutate `argv` because that can create subtle bugs. You can work around this by first copying all of its properties and working with that variable:
````js
const newArgv = {...argv};
for (const option in configObject) {
  // ...
  newArgv[option] = configObject[option];
}
// ...
return newArgv;
````